src/storage/uci/mod.rs

```diff
@@ -0,0 +1,418 @@
+mod uci_config;
+mod uci_option;
+mod uci_section;
+
+use crate::utils::Error;
+use smol::fs::unix::PermissionsExt;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::prelude::*;
+use std::path::Path;
+use std::sync::Mutex;
+use tempfile::NamedTempFile;
+
+use uci_config::UciConfig;
+use uci_option::UciOption;
+use uci_section::UciSection;
+
+use self::uci_option::UciOptionType;
+
+pub struct UciTree {
+    dir: Box<Path>,
```

Comment:
the dir maybe not need to be wrapped in Box

Reply:
because path does not implement 'Sized' trait